command/agent/agent_endpoint_test.go

```diff
@@ -2007,13 +1706,8 @@ func TestAgent_Monitor_ACLDeny(t *testing.T) {
 	defer srv.agent.Shutdown()
 
 	// Try without a token.
-	req, err := http.NewRequest("GET", "/v1/agent/monitor", nil)
-	if err != nil {
-		t.Fatalf("err: %v", err)
-	}
-
-	_, err = srv.AgentMonitor(nil, req)
-	if err == nil || !strings.Contains(err.Error(), permissionDenied) {
+	req, _ := http.NewRequest("GET", "/v1/agent/monitor", nil)
+	if _, err := srv.AgentMonitor(nil, req); err != errPermissionDenied {
```

Comment:
Actually, that was my first attempt of eliminating the `strings.Contains(err.Error(), "permission denied")` but that compares identity of the errors and I'm not sure if we are always getting the same `errPermissionDenied` here or if this is one that got bubbled up from the RPC layer or embedded in some other error message. Therefore, I stuck with the string comparison. If we want to change that then we only need to change it in one place or we can replace the helper with this code.